### PR TITLE
DOC Explain how to customise image rendering

### PR DESCRIPTION
**Pull request by maxime-rainville**
## Parent issue
https://github.com/silverstripe/.github/issues/230

### CODE REVIEW

File: en/02_Developer_Guides/14_Files/02_Images.md [hunk 1]
```diff
@@ -334,6 +334,20 @@ SilverStripe\Core\Injector\Injector:
 
 Manipulated images are stored as "file variants" in the same folder structure as the original image. The storage mechanism is described in the ["File Storage" guide](file_storage).
 
+## Controlling how images are rendered
+
+Developers can customise how `Image` instances are rendered on their website by overriding the the `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` template file.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/14_Files/02_Images.md]:
```suggestion
Developers can customise how `Image` instances are rendered on their website by overriding the `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` template file.
```

File: en/02_Developer_Guides/14_Files/02_Images.md [hunk 2]
```diff
@@ -334,6 +334,20 @@ SilverStripe\Core\Injector\Injector:
...
+
+This will apply to images added to an HTMLEditorField and SS template images.
+
+You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.
+
+- Add a `SilverStripe/Assets/Shortcodes/ImageShortcodeProvider_Image.ss` to your theme to control images added to an HTMLEditorField.
+- Add a `DBFile_Image.ss` file to the root of your theme to control only SS templates images.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/14_Files/02_Images.md]:
```suggestion
- Add a `DBFile_Image.ss` file to the root of your theme to control only SS template images.
```

File: en/02_Developer_Guides/14_Files/02_Images.md [hunk 3]
```diff
@@ -334,6 +334,20 @@ SilverStripe\Core\Injector\Injector:
 
 Manipulated images are stored as "file variants" in the same folder structure as the original image. The storage mechanism is described in the ["File Storage" guide](file_storage).
 
+## Controlling how images are rendered
+
+Developers can customise how `Image` instances are rendered on their website by overriding the the `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` template file.
+
+This will apply to images added to an HTMLEditorField and SS template images.
+
+You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/14_Files/02_Images.md]:
```suggestion
You can also choose to have different rendering logic for HTMLEditorField images and for SS template images by overriding different templates.
```

File: en/08_Changelogs/5.3.0.md [hunk 4]
```diff
@@ -51,6 +52,26 @@ Out of the box `silverstripe/assets` provides the new [`InterventionImageFileCon
 
 See [convert a file to a different format](/developer_guides/files/file_manipulation/#file-conversion) for more details.
 
+### Improve customisability of rendered images {#image-rendering}
+
+We've aligned how images added to an HTMLEditorField and SS template images are rendered. The markup up for both `<img>` tag is now generated from the same template file. This will make it easier to customise how images are rendered on your site.
+
+Simply add a `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` file to your Silverstripe CMS theme.
```
Comment (emteknetnz) [inline en/08_Changelogs/5.3.0.md]:
```suggestion
We've aligned how images added to an [`HTMLEditorField`](api:SilverStripe\Forms\HTMLEditor\HTMLEditorField) and images inside SS templates are rendered. The markup up for both `<img>` tags are now generated from the same template file. This makes it easier to customise how images by creating a `templates/SilverStripe/Assets/Storage/DBFile_Image.ss` file to your Silverstripe CMS theme.
```

File: en/08_Changelogs/5.3.0.md [hunk 5]
```diff
@@ -51,6 +52,26 @@ Out of the box `silverstripe/assets` provides the new [`InterventionImageFileCon
...
+
+This opens many interesting possibilities. For example, this code snippet will provide a WebP version of any image rendered on your site.
+
+```html
+<picture>
+    <source srcset="{$Convert('webp').Link}" type="image/webp">
```
Comment (emteknetnz) [inline en/08_Changelogs/5.3.0.md]:
```suggestion
    <source srcset="<% $Convert('webp').Link %>" type="image/webp">
```
Can you use those brackets in templates? I would have thought you'd need the regular template "brackets"

File: en/08_Changelogs/5.3.0.md [hunk 6]
```diff
@@ -51,6 +52,26 @@ Out of the box `silverstripe/assets` provides the new [`InterventionImageFileCon
...
+    <img $AttributesHTML >
+</picture>
+```
+
+You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.
+
+- Add a `SilverStripe/Assets/Shortcodes/ImageShortcodeProvider_Image.ss` to your theme to control images added to an HTMLEditorField.
+- Add a `DBFile_Image.ss` file to the root of your theme to control only SS templates images.
```
Comment (emteknetnz) [inline en/08_Changelogs/5.3.0.md]:
```suggestion
You can also choose to have different rendering logic for HTMLEditorField images and for SS template images by overriding different templates.

- Add a `SilverStripe/Assets/Shortcodes/ImageShortcodeProvider_Image.ss` to your theme to control images added to an HTMLEditorField.
- Add a `DBFile_Image.ss` file to the root of your theme to control only SS template images.
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

**Review by emteknetnz — CHANGES_REQUESTED** [review]
There's also a linting failure

File: en/08_Changelogs/5.3.0.md (hunk 5 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/5.3.0.md]:
```suggestion
    <source srcset="$Convert('webp').Link" type="image/webp">
```
This is more standard.
The suggestion to use `<%` would give you an error.

File: en/08_Changelogs/5.3.0.md (hunk 4 above)
Comment (ChloeSartorelli) [inline en/08_Changelogs/5.3.0.md]: Sounds like there's no built-in template. Try something like this:
```suggestion
The template for this is `templates/SilverStripe/Assets/Storage/DBFile_Image.ss`. Add a file with that path to your Silverstripe CMS theme if you want to override it.
```

File: en/08_Changelogs/5.3.0.md [hunk 7]
```diff
@@ -51,6 +52,26 @@ Out of the box `silverstripe/assets` provides the new [`InterventionImageFileCon
 
 See [convert a file to a different format](/developer_guides/files/file_manipulation/#file-conversion) for more details.
 
+### Improve customisability of rendered images {#image-rendering}
+
+We've aligned how images added to an HTMLEditorField and SS template images are rendered. The markup up for both `<img>` tag is now generated from the same template file. This will make it easier to customise how images are rendered on your site.
```
Comment (ChloeSartorelli) [inline en/08_Changelogs/5.3.0.md]: I assume when you say "SS template" you mean "when you invoke the `forTemplate()` method for an image" i.e. `$MyImage`.

Should probably be more explicit about that, especially in the `02_images.md` file. It's very common for people to write explicit `img` tags for each use of an image in a template, so the distinction is important.

File: en/08_Changelogs/5.3.0.md [hunk 8]
```diff
@@ -51,6 +52,26 @@ Out of the box `silverstripe/assets` provides the new [`InterventionImageFileCon
...
+This opens many interesting possibilities. For example, this code snippet will provide a WebP version of any image rendered on your site.
+
+```html
+<picture>
+    <source srcset="{$Convert('webp').Link}" type="image/webp">
+    <img $AttributesHTML >
+</picture>
+```
+
+You can also choose to have different rendering logic for HTMLEditorField images and for SS templates images by overriding different templates.
```
Comment (ChloeSartorelli) [inline en/08_Changelogs/5.3.0.md]:
```suggestion
You can also choose to have different rendering logic for `HTMLEditorField` images and for SS templates images by overriding different templates.
```

Comment (ChloeSartorelli) [inline en/08_Changelogs/5.3.0.md]:
```suggestion
You can also choose to have different rendering logic for HTMLEditorField images and for images invoked in templates by overriding different templates.
```
I don't think anywhere else in the docs we say "SS templates"

File: en/02_Developer_Guides/14_Files/02_Images.md (hunk 1 above)
Comment (maxime-rainville) [reply en/02_Developer_Guides/14_Files/02_Images.md]: Fixed

File: en/02_Developer_Guides/14_Files/02_Images.md (hunk 2 above)
Comment (maxime-rainville) [reply en/02_Developer_Guides/14_Files/02_Images.md]: I used Guy's "invoked in templates" language instead.

File: en/02_Developer_Guides/14_Files/02_Images.md (hunk 3 above)
Comment (maxime-rainville) [reply en/02_Developer_Guides/14_Files/02_Images.md]: I used Guy's "invoked in templates" language instead.

File: en/08_Changelogs/5.3.0.md (hunk 4 above)
Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: I did the first paragraph fix. Not the second.

File: en/08_Changelogs/5.3.0.md (hunk 5 above)
Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: I took Guy's suggestion.

File: en/08_Changelogs/5.3.0.md (hunk 6 above)
Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: I used Guy's "invoked in templates" language instead.

File: en/08_Changelogs/5.3.0.md (hunk 4 above)
Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: Done

File: en/08_Changelogs/5.3.0.md (hunk 7 above)
Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: I adopted the "invoked in templates" language.

File: en/08_Changelogs/5.3.0.md (hunk 8 above)
Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: Done

Comment (maxime-rainville) [reply en/08_Changelogs/5.3.0.md]: Done.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**